### ISSUE TITLE
Type Error Occurs When Passing i18nKey without Context to Trans Component in TypeScript

### ISSUE THREAD

**Issue by tachibanayu24** [i18next/react-i18next#1688]
## 🐛 Bug Report

Hi,

When using a key-value with context as props.i18nKey in the Trans component, a TypeScript type error occurs. It seems to be inferred without problems with the t() function.

<img width="938" alt="Screenshot 2023-10-14 at 19 50 17" src="https://github.com/i18next/i18next/assets/23476543/11d77ed0-2cee-416e-860e-d496c0545583">

If this is not a bug and is an issue that can be addressed by some means, I apologize.

## To Reproduce

* survey.json
```json
{
  "sample_A": "A context",
  "sample_B": "B context"
}
```

* i18n.d.ts
```ts
import "i18next";
import survey from "survey.json";

declare module "i18next" {
  interface CustomTypeOptions {
    resources: {
      survey: typeof survey;
    };
  }
}
```

* config.ts
```ts
import i18next from "i18next";
import { initReactI18next } from "react-i18next";

import enSurvey from "./en/survey.json";
import jaSurvey from "./ja/survey.json";

i18next.use(initReactI18next).init({
  fallbackLng: "ja",
  interpolation: {
    escapeValue: false,
  },
  returnObjects: true,
  resources: {
    ja: {
      survey: jaSurvey,
    },
    en: {
      survey: enSurvey,
    },
  },
});

```

* jsx
```ts
{/* This is inferred as `sample_A | sample_B`. */}
<Trans ns="survey" i18nKey="sample" context="A" />
{/* /Seems to be correctly inference */}
{t("sample", { ns: 'survey', context: "A" })}
```

## Expected behavior

Trans.props.i18nKey should be inferred as `sample | sample_A | sample_B`.

## Your Environment

- *runtime version*: node v18.12.0
- *i18next version*: 23.5.1 
- *react-i18next version*: 13.3.0
- *os*: Mac

**Comment by adrai** [comment]
@tachibanayu24 feel free to try a PR to address this

**Comment by adrai** [comment]
btw: the Trans property is not content but context

**Comment by adrai** [comment]
alternatively as a workaround you can use: `<Trans ns="survey" i18nKey="sample" tOptions={{ context: 'A' }} />`

**Comment by tachibanayu24** [comment]
Oh, my apologies. I've corrected 'content' to 'context' 🙏 

I attempted to submit a PR regarding this issue, but I must admit, I was challenged by the intricacies of the code. It’s evident that the library expertly encapsulates complexity, providing a beautifully straightforward user experience concerning types. I apologize for not being able to contribute directly in this instance.

Thank you for providing the workaround. I’m contemplating whether to proceed with this approach or to explore an alternative method, possibly avoiding the use of context and opting to nest objects, managing variances with template literals on the user side of the Trans component.
